tests/trainer/logging_/test_train_loop_logging.py

```diff
@@ -261,7 +261,9 @@ def __init__(self):
 
         def training_step(self, batch, batch_idx, hiddens):
             assert hiddens == self.test_hidden, "Hidden state not persistent between tbptt steps"
-            self.test_hidden = torch.rand(1)
+            if hiddens is not None:
+                assert hiddens.grad_fn is None
+            self.test_hidden = torch.tensor(2., requires_grad=True).pow(2)
```

Comment:
great! verified it fails without the fix. 
either it was there forever or I had broken it with loop refactor :(((